lib/spack/spack/test/spec_yaml.py

```diff
@@ -190,11 +190,10 @@ def test_ordered_read_not_required_for_consistent_dag_hash(
             reversed_json_string
         )
 
-        # TODO: remove this when build deps are in provenance.
-        spec = spec.copy(deps=('link', 'run'))
```

Comment:
with the refactor above we'd bring back this line and make it:

```python
spec = spec.copy(deps=hash_type.deptype)
```

so that we can round-trip all three hash types.

Reply:
We can't compare the full hashes below if we do that.